src/fromDelta.js

```diff
@@ -1,148 +1,278 @@
+  }
+  if (italic) {
+    inline = u("emphasis", [inline]);
+  }
+  if (entity.type.toLowerCase() === "link") {
+    inline = u("link", { url: entity.data.url, title: null }, [inline]);
+  }
+  return inline;
+};
+
+function handleImageUrl(url) {
+  if (url.indexOf("&") > 0) url = url.substring(0, url.indexOf("&"));
+  return encodeURI(url);
+}
+
+const defaultConvertLine = ({ type, data }) => {
+  switch (type) {
+    case "unstyled": {
+      return {
+        type: "unstyled"
+      };
+    }
+    case "header-one": {
+      return {
+        type: "heading",
+        depth: 1
+      };
+    }
+    case "header-two": {
+      return {
+        type: "heading",
+        depth: 2
+      };
+    }
+    case "unordered-list-item": {
+      return {
+        type: "listItem",
+        checked: null,
+        depth: data ? data.depth || 0 : 0,
+        ordered: false
+      };
+    }
+    case "ordered-list-item": {
+      return {
+        type: "listItem",
+        checked: null,
+        depth: data ? data.depth || 0 : 0,
+        ordered: true
+      };
+    }
+    case "todo-block": {
+      return {
+        type: "listItem",
+        checked: data.checked === undefined ? false : data.checked,
+        depth: data ? data.depth || 0 : 0,
+        ordered: false
+      };
+    }
+    case "image": {
+      return {
+        type: "image",
+        url: handleImageUrl(data.url)
+      };
+    }
+    case "separator": {
+      return {
+        type: "separator"
+      };
+    }
+    case "table-cell": {
+      return {
+        type: "tableCell",
+        rowBreak: data.rowBreak
+      };
+    }
+    case "code-block": {
+      return {
+        type: "code"
+      };
+    }
+  }
+  return {
+    type: "paragraph"
+  };
+};
+
+const insertListInChildren = children => {
+  let lists = [];
+  let listItems = [];
+  let others = [];
+  let style = { ordered: -1 };
+  children.forEach(child => {
+    if (child.type === "listItem") {
+      //  checking if the list at the same depth is of the same type.
+      //  if it is, we will use its type, otherwise, a new one is created
+      if (style.ordered !== -1 && child.ordered !== style.ordered) {
+        lists = lists.concat([listItems]);
+        listItems = [];
+      }
+      listItems = listItems.concat(child);
+      style = { ordered: child.ordered, checked: child.checked };
+    } else {
+      others = others.concat(child);
+    }
+  });
+  lists = lists.concat([listItems]);
+  lists.map(list => {
+    if (list.length) {
+      others = others.concat(
+        u("list", { ordered: list[0].ordered, start: 1 }, list)
+      );
+    }
+  });
+  return others;
+};
+
+function closeUpList(currentNode) {
+  if (currentNode) {
+    while (currentNode.depth > 0) {
+      let parentNode = currentNode.parent();
+      parentNode.children = insertListInChildren(parentNode.children);
+      currentNode = currentNode.parent();
+    }
+    currentNode.parent().children = insertListInChildren(
```

Comment:
Since we don't know what the method `parent()` does exactly, it would be better to affect its result to a var if we are going to use it more than once, for performance reasons.